docs/data_types/tensor/tensor.md

```diff
@@ -0,0 +1,118 @@
+# Tensor
+
+
+
+DocArray support several tensor type that can be used inside BaseDoc. 
+
+The main ones are : 
+- [`NdArray`][docarray.typing.tensor.NdArray] for numpy tensor
+- [`TorchTensor`][docarray.typing.tensor.TorchTensor] for Pytorch tensor
+- [`TensorFlowTensor`][docarray.typing.tensor.TensorFlowTensor] for tensorflow tensor
+
+The three of them inherit from their respective framework tensor type. This means that they can be used natively inside their framework.
+
+We also have a [`AnyTensor`][docarray.typing.tensor.AnyTensor] that is the Union of the three previous tensor type. 
+This is a generic placeholder to specify that it can work with any tensor type (numpy, torch, tensorflow).
+
+## Tensor typed validation
+
+You don't need to instantiate directly the  [`NdArray`][docarray.typing.tensor.NdArray] , [`TorchTensor`][docarray.typing.tensor.TorchTensor], [`TensorFlowTensor`][docarray.typing.tensor.TensorFlowTensor] directly by yourself. They are only use as type hint field fot the [`BaseDoc`][docarray.base_doc.doc.BaseDoc] validation.
+[`BaseDoc`][docarray.base_doc.doc.BaseDoc] will cast the native tensor type into the respective DocArray tensor type.
+
+Let's take an example 
+
+```python
+from docarray import BaseDoc
+from docarray.typing import NdArray
+
+import numpy as np
+
+
+class MyDoc(BaseDoc):
+    tensor: NdArray
+
+
+doc = MyDoc(tensor=np.zeros(100))
+
+assert isinstance(doc.tensor, NdArray)  # True
+``` 
+here you see that the `doc.tensor` is a `NdArray`. 
+
+```python
+assert isinstance(doc.tensor, np.ndarray)  # True as well
+``` 
+
+But since it inherits from `np.ndarray` you can use it as a numpy array.
```

Comment:
```suggestion
But since it inherits from `np.ndarray` you can also use it as a normal numpy array. The same holds for pytorch and `TorchTensor`.
```